Fix compiler warnings

garbear opened the pull request:
This PR contains all the compiler warning fixes I've stashed away since last release.

I measured the warning count decrease on Windows. We went from 1,030 warnings to 978. Improvement!

## How Has This Been Tested?
Compile-tested on Windows 7 x64 and OSX.

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

garbear committed:
guilib: Silence compiler warning

Warning was:

xbmc/guilib/DDSImage.cpp: In member function 'bool CDDSImage::ReadFile(const string&)':
xbmc/guilib/DDSImage.cpp:111:44: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (file.Read(m_data, m_desc.linearSize) != m_desc.linearSize)
                                            ^

garbear committed:
Silence compiler warning in Picture.cpp

Warning was:

xbmc/pictures/Picture.cpp: In static member function 'static bool CPicture::CreateThumbnailFromSurface(const unsigned char*, int, int, int, const string&)':
xbmc/pictures/Picture.cpp:93:87: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   const bool ret = file.OpenForWrite(thumbFile, true) && file.Write(thumb, thumbsize) == thumbsize;
                                                                                       ^

garbear committed:
Silence compiler warning in AESinkAUDIOTRACK.cpp

Warning was:

xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp: In member function 'int CAESinkAUDIOTRACK::AudioTrackWrite(char*, int, int)':
xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:214:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (m_charbuf.size() != (sizeInBytes - offsetInBytes))
                          ^

garbear committed:
utils: Fix compiler warning

Warning was:

Mime.cpp: In function 'std::map<std::basic_string<char>, std::basic_string<char> > fillMimeTypes()':
Mime.cpp:32:36: note: variable tracking size limit exceeded with -fvar-tracking-assignments, retrying without
 std::map<std::string, std::string> fillMimeTypes()
                                    ^

garbear committed:
Peripherals: Fix compiler warning on Windows

garbear committed:
Games: Fix compiler warnings

garbear committed:
[OSX]XBMCHelper: Silence compiler warning

Warning was:

XBMCHelper.m:219:10: warning: 8 enumeration values not handled in switch

garbear committed:
cdrip: Silence compiler warning

Warning was:

CDDARipJob.cpp:118:10: warning: ‘encoder’ may be used uninitialized in this function [-Wmaybe-uninitialized]

   delete encoder;
          ^

garbear committed:
Silence compiler warning in DVDSubtitleTagMicroDVD.cpp

Warning was:

DVDSubtitleTagMicroDVD.cpp:50:35: warning: ‘pos3’ may be used uninitialized in this function [-Wmaybe-uninitialized]
           strUTF8.erase(pos, pos3 - pos + 1);
                                   ^

Rechi commented on lib/libUPnP/Platinum/Source/Core/PltCtrlPoint.cpp:
```diff
@@ -702,7 +702,7 @@ PLT_CtrlPoint::DecomposeLastChangeVar(NPT_List<PLT_StateVariable*>& vars)
             NPT_XmlElementNode* instance = NULL;
             for (NPT_Cardinal i=0; i<node->GetChildren().GetItemCount(); i++) {
-                NPT_XmlElementNode* child;
+                NPT_XmlElementNode* child = nullptr;
```
needs a patch
If nothing is broken I wouldn't change it as it isn't our lib.

Rechi commented on lib/libUPnP/Platinum/Source/Core/PltDeviceHost.cpp:
```diff
@@ -494,7 +494,7 @@ PLT_DeviceHost::ProcessHttpPostRequest(NPT_HttpRequest&              request,
     PLT_Service*              service;
     NPT_XmlElementNode*       soap_body;
-    NPT_XmlElementNode*       soap_action;
+    NPT_XmlElementNode*       soap_action = nullptr;
```
same

Rechi commented on tools/EventClients/Clients/OSXRemote/XBMCHelper.m:
```diff
@@ -253,6 +253,8 @@ - (NSString *)buttonNameForButtonCode:(HIDRemoteButtonCode)buttonCode
 		case kHIDRemoteButtonCodeMenuHold:
 			return (@"Menu (hold)");
+      break;
+        default:
```
Why not mention the missing cases instead of default?
Last time I looked at it, there were cases which should be handled by this switch statement.

garbear replied:
I don't know objective-C and have never used this helper. Can you take care of this?

garbear replied:
@Memphiz or @FernetMenta could you add the 8 missing labels?

Rechi replied:
`kHIDRemoteButtonCodePlay` and `kHIDRemoteButtonCodePlayPause` should be handled. For the other cases you can leave the default case.

garbear replied:
I added `kHIDRemoteButtonCodePlay`, but it seems `kHIDRemoteButtonCodePlayPause` is already handled?

Rechi replied:
my fault mixed something up, correct ones are `kHIDRemoteButtonCodePlay` and `kHIDRemoteButtonCodePlayHold`

garbear commented:
dropped libUPNP commit

garbear committed:
EmbeddedArtInfo: Fix missing virtual destructor

Warning was:

memory:1814:94: warning: destructor called on non-final 'EmbeddedArtInfo' that has virtual functions but non-virtual destructor [-Wdelete-non-virtual-dtor]

garbear committed:
Addons: Fix compiler warning

Warning was:

General.cpp:264:10: warning: comparison of two values with different enumeration types in switch statement ('QueueMsg' and 'ADDON::queue_msg') [-Wenum-compare-switch]
    case QUEUE_INFO:
         ^~~~~~~~~~

garbear committed:
LangInfo.cpp: Fix compiler warning

Warning was:

warning: the specified comparator type does not provide a const call operator [-Wuser-defined-warnings]

garbear committed:
DPMSSupport.cpp: Fix compiler warning

Warning was:

DPMSSupport.cpp:25:12: warning: comparison of unsigned enum expression < 0 is always false [-Wtautological-unsigned-enum-zero-compare]
  if (mode < 0 || mode > NUM_MODES)
      ~~~~ ^ ~

garbear committed:
XBDateTime.h: Fix compiler warning

Warning was:

memory:1814:94: warning: destructor called on non-final 'CDateTime' that has virtual functions but non-virtual destructor [-Wdelete-non-virtual-dtor]

garbear commented:
Added five more compiler warning fixes, including two (0c4ba83 and 1b13be5) introduced by https://github.com/xbmc/xbmc/pull/14033

Rechi commented:
jenkins build this please
